### PR TITLE
Simplify napatech v10

### PR DESCRIPTION
**Pull request by pjyoung1**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/2766

Describe changes:

Added functionality to simplify integration with Napatech cards.

* There is now an option to automatically create streams on the correct NUMA node when
  using cpu affinity.
* When not using cpu affinity the user can specify streams to be created in the suricata.yaml file.
  It is no longer required to use NTPL to create streams before running suricata.
* The legacy usage model of running NTPL to create streams is still available. This can be
  used for legacy configurations and complex configurations that cannot be satisfied by the
  auto-config option.

Also addresses review comments from PR [simplify_napatech_v6](https://github.com/OISF/suricata/pull/3635)

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output (if applicable):

### CODE REVIEW

File: doc/userguide/capture-hardware/napatech.rst [hunk 1]
```diff
@@ -110,105 +74,250 @@ prepare for compilation::
...
-of the packets.  If this is enabled without another application sharing the stream it will result in sub-optimal packet
-buffering.
-
+Suricata configuration
+----------------------
+
+Now edit the suricata.yaml file to configure the system. There are three ways
+the system can be configured:
+
+     1. Auto-config without cpu-affinity: In this mode you specify the stream
```
Comment (norg) [inline doc/userguide/capture-hardware/napatech.rst]: unnecessary space in front of 1.

File: doc/userguide/capture-hardware/napatech.rst [hunk 2]
```diff
@@ -110,105 +74,250 @@ prepare for compilation::
...
+
+Now edit the suricata.yaml file to configure the system. There are three ways
+the system can be configured:
+
+     1. Auto-config without cpu-affinity: In this mode you specify the stream 
+    configuration in suricata.yaml file and allow the threads to 
+    roam freely. This is good for single processor systems where NUMA node 
+    configuration is not a performance concern.
+  
+    2. Auto-config with cpu-affinity:  In this mode you use the cpu-affinity
```
Comment (norg) [inline doc/userguide/capture-hardware/napatech.rst]: double spaces after :

File: doc/userguide/capture-hardware/napatech.rst [hunk 3]
```diff
@@ -228,8 +337,102 @@ In addition to counters host buffer utilization is tracked and logged.  This is
...
+
+Make sure that there are enough host-buffers declared in ntservice.ini to 
+accommodate the number of cores/streams being used.
+
 Support
 -------
 
 Contact a support engineer at: ntsupport@napatech.com
 
+Napatech Documentation can be found at: https://docs.napatech.com (CLick the search icon, with no search text,
```
Comment (norg) [inline doc/userguide/capture-hardware/napatech.rst]: C**L** should be C**l**

File: suricata.yaml.in [hunk 4]
```diff
@@ -1805,19 +1805,63 @@ napatech:
...
-    # use_all_streams set to "yes" will query the Napatech service for all configured
-    # streams and listen on all of them. When set to "no" the streams config array
-    # will be used.
-    use-all-streams: yes
-
-    # The streams to listen on.  This can be either:
-    #   a list of individual streams (e.g. streams: [0,1,2,3])
+    #hba: -1
+
+    # When use_all_streams is set to "yes" the initialization code will query
```
Comment (norg) [inline suricata.yaml.in]: I would add '' to config options like 'use_all_streams' to emphasize them.

**Comment by norg** [comment]
Doc looks good but quite a bit long, could you see if you can reduce the examples?

Also several minor fixes need, like removing double spaces.

File: src/util-napatech.c [hunk 5]
```diff
@@ -343,7 +323,74 @@ static bool RegisteredStream(uint16_t stream_id, uint16_t num_registered,
     return false;
 }
 
-uint16_t NapatechGetStreamConfig(NapatechStreamConfig stream_config[])
+static uint32_t CountWorkerThreads(void)
+{
+    int worker_count = 0;
+
+    ConfNode *root = ConfGetNode("threading.cpu-affinity");
```
Comment (victorjulien) [inline src/util-napatech.c]: if this fails line 333 below will be a null ptr deref

**Comment by victorjulien** [comment]
Btw @norg I don't agree the doc is too long. I'm fine with keeping it as-is and just fix the smaller comments you had.

**Comment by victorjulien** [comment]
Please also close the old PR and link to it from the new PR.

**Pull request closed without merging**